src/ArticlesList.jsx

```diff
@@ -1,29 +1,39 @@
-import { useState, useEffect } from "react"
-import getArticles from './utils/newsApi'
-import Articles from "./Articles"
+import { useState, useEffect } from "react";
+import getArticles from './utils/newsApi';
+import Articles from "./Articles";
 
+import ArticlesByTopic from "./ArticlesByTopic";
+import Navbar from "./TopicsNavBar";
 
 export default function ArticlesList(){
 
-    const [articles, setArticles] = useState([])
+    const [articles, setArticles] = useState([]);
+    const [selectedTopic, setSelectedTopic] = useState(null);
 
     useEffect(()=>{
         getArticles()
         .then((response)=>{
-            setArticles(response)
+            setArticles(response);
         })
         .catch((err)=>{
             console.log('Error:', err);
-        })
-    }, [])
+        });
+
+    }, []);
 
+    const topics = [...new Set(articles.map(article => article.topic))];
```

Comment:
This isn't how you should be getting topics. We built an endpoint specifically for this